plugins/vppv2/aclplugin/aclplugin.go

```diff
@@ -0,0 +1,116 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+//go:generate descriptor-adapter --descriptor-name Acl --value-type *acl.Acl --meta-type *aclidx.AclMetadata --import "aclidx" --import "../model/acl" --output-dir "descriptor"
+///go:generate descriptor-adapter --descriptor-name AclInterface --value-type *acl.Acl_Interface --import "aclidx" --import "../model/acl" --output-dir "descriptor"
```

Comment:
@milanlenco from my understanding I cannot do this without another descriptor for the ACL-to-interface assignment, right? Because of the binary API calls required. However when I checked how you did bd-to-interface assignment, you used proto message BD_Interface, which does not seem applicable here, because interface assigments for ACLs are defined inside string array which is not proto message. Does the model need to be adjusted so it defined proto message for each interface assignment or can it be done with the prototypes.Empty?

Reply:
Correct, you have two options:
1. create plugin-internal proto model for interface assignment, or, easier
2. use empty value and derive key that contains all neccessary information (ACL name, interface, + different prefix for ingress/egress): you will need to put all this information into the key anyway to make it unique

The first approach I have already used for DHCP client: https://github.com/ligato/vpp-agent/blob/dev/plugins/vppv2/ifplugin/descriptor/dhcp.go#L113-L164  (there it was just the interface name that was needed)